src/main/java/com/thullo/service/BoardServiceImpl.java

```diff
@@ -171,13 +190,12 @@ private String generateThreeLetterWord(String boardName) {
                 return threeLetterWord;
             }
         }
-
         throw new IllegalStateException("All three-letter substrings have been used. Please choose a different board name.");
     }
 
-    public boolean hasBoardRole(String boardOwner, String boardTag) {
+    public boolean hasBoardRole(String boardOwner, String boardTag) throws BadRequestException {
```

Comment:
If the previous comment is taken into consideration, this implementation will work well as this approach is not designed to throw an exception.